src/main/java/de/hpi/bpt/argos/api/response/ResponseFactoryImpl.java

```diff
@@ -152,6 +163,49 @@ public String getEvent(long eventId) {
 	}
 
 	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void createEventType(String requestBody) {
+		try {
+			JsonObject jsonBody = jsonParser.parse(requestBody).getAsJsonObject();
+
+			String eventQuery = jsonBody.get(JSON_EVENT_SUBSCRIPTION_QUERY_ATTRIBUTE).getAsString();
+			JsonObject jsonEventType = jsonBody.get(JSON_EVENT_TYPE_ATTRIBUTE).getAsJsonObject();
+
+			if (eventQuery == null || eventQuery.length() == 0) {
+				halt(RestInputValidationService.getHttpErrorCode(), "no event query found");
```

Comment:
This does not request any existing data, thus 404 is not a good response code. But I admit, that the error message should be changed.